certdepot/mongo_depot.go

```diff
@@ -0,0 +1,234 @@
+package certdepot
+
+import (
+	"time"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"github.com/square/certstrap/depot"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type User struct {
+	ID            string    `bson:"_id"`
+	Cert          string    `bson:"cert"`
+	PrivateKey    string    `bson:"private_key"`
+	CertReq       string    `bson:"cert_req"`
+	CertRevocList string    `bson:"cert_revoc_list"`
+	TTL           time.Time `bson:"ttl"`
+}
+
+var (
+	userIDKey            = bsonutil.MustHaveTag(User{}, "ID")
+	userCertKey          = bsonutil.MustHaveTag(User{}, "Cert")
+	userPrivateKeyKey    = bsonutil.MustHaveTag(User{}, "PrivateKey")
+	userCertReqKey       = bsonutil.MustHaveTag(User{}, "CertReq")
+	userCertRevocListKey = bsonutil.MustHaveTag(User{}, "CertRevocList")
+	userTTLKey           = bsonutil.MustHaveTag(User{}, "TTL")
+)
+
+type mongoCertDepot struct {
+	session        *mgo.Session
+	databaseName   string
+	collectionName string
+	expireAfter    time.Duration
+}
+
+type MgoCertDepotOptions struct {
+	MongoDBURI           string
+	MongoDBDialTimeout   time.Duration
+	MongoDBSocketTimeout time.Duration
+	DatabaseName         string
+	CollectionName       string
+	ExpireAfter          time.Duration
+}
+
+// Create a new cert depot in the specified MongoDB.
+func NewMgoCertDepot(opts MgoCertDepotOptions) (depot.Depot, error) {
+	return newMgoCertDeposit(nil, opts)
+}
+
+// Create a new cert depot in the specified MongoDB, using an existing session.
+func NewMgoCertDepotWithSession(s *mgo.Session, opts MgoCertDepotOptions) (depot.Depot, error) {
+	return newMgoCertDeposit(s, opts)
+}
+
+func validate(opts MgoCertDepotOptions) MgoCertDepotOptions {
```

Comment:
make this a method on opts.

Reply:
it fails with lint if I do that... I get an error that the use of ops is ineffectual:
certdepot/mongo_depot.go:61:8:warning: ineffective assignment to field MongoDBURI (SA4005) (megacheck)
certdepot/mongo_depot.go:64:8:warning: ineffective assignment to field MongoDBDialTimeout (SA4005) (megacheck)
certdepot/mongo_depot.go:67:8:warning: ineffective assignment to field MongoDBSocketTimeout (SA4005) (megacheck)
certdepot/mongo_depot.go:70:8:warning: ineffective assignment to field DatabaseName (SA4005) (megacheck)
certdepot/mongo_depot.go:73:8:warning: ineffective assignment to field CollectionName (SA4005) (megacheck)
certdepot/mongo_depot.go:76:8:warning: ineffective assignment to field ExpireAfter (SA4005) (megacheck)
--- FAIL: lint-certdepot (26.153491873s)

Reply:
oh right, duh